packages/core/src/tile/model/point/pointContent.ts

```diff
@@ -0,0 +1,327 @@
+import {
+  CreateBox,
+  CreateSphere,
+  Mesh,
+  Nullable,
+  Scene,
+  ShaderMaterial,
+  UniformBuffer,
+  VertexBuffer,
+  VertexData
+} from '@babylonjs/core';
+import { TileContent, TileUpdateOptions } from '../tileContent';
+import { Tile } from '../tile';
+import { PointDataContent } from '../../../types';
+import {
+  PointCloudMaterial,
+  PointCloudMaterialWebGPU
+} from '../../materials/pointShaderMaterial';
+import { Feature } from '@tiledb-inc/viz-common';
+import { FeatureType } from '@tiledb-inc/viz-common';
+import { PointShape } from '@tiledb-inc/viz-common';
+import { TypedArray } from '../../types';
+
+type PointCloudData = {
+  position: Float32Array;
+  attributes: Record<string, TypedArray>;
+};
+
+export type PointCloudUpdateOptions = TileUpdateOptions & {
+  data?: PointCloudData;
+  UBO?: UniformBuffer;
+  FrameUBO?: UniformBuffer;
+  feature?: Feature;
+  pointSize?: number;
+  pointShape?: PointShape;
+};
+
+export class PointTileContent extends TileContent {
+  private pointCount: number;
+  private material: Nullable<ShaderMaterial>;
+
+  constructor(scene: Scene, tile: Tile<PointDataContent, PointTileContent>) {
+    super(scene, tile);
+
+    this.pointCount = 0;
+    this.material = null;
+  }
+
+  public update(options: PointCloudUpdateOptions): void {
+    super.update(options);
+
+    if (this.scene.getEngine().isWebGPU) {
+      this.updateWebGPU(options);
+    } else {
+      this.updateWebGL(options);
+    }
+  }
+
+  private onDataUpdateWebGPU(data: PointCloudData) {
+    this.pointCount = data.position.length / 3;
+    this.buffers = data.attributes;
+
+    if (this.meshes.length === 0) {
+      console.error('Mesh is unitiliazed for point cloud tile');
+    }
+
+    this.meshes[0]._thinInstanceDataStorage.instancesCount = this.pointCount;
+    this.meshes[0].layerMask = this.tile.mask;
+    this.meshes[0].setVerticesBuffer(
+      new VertexBuffer(this.scene.getEngine(), data.position, 'loc', {
+        size: 3,
+        stride: 3,
+        instanced: true
+      })
+    );
+  }
+
+  private onDataUpdateWebGL(data: PointCloudData) {
+    this.pointCount = data.position.length / 3;
+
+    if (this.meshes.length === 0) {
+      this.material = PointCloudMaterial(this.scene);
+
+      this.meshes.push(new Mesh(this.tile.id.toString(), this.scene));
+      this.meshes[0].setBoundingInfo(this.tile.boundingInfo);
+      this.meshes[0].material = this.material;
```

Comment:
lots of references going on here -- risk of mutation-by-caller unless i'm mistaken -- ?

Reply:
For bounding info there is no risk since the bounding box of the mesh should change if the tiles bounding box changes. For the material the reference passing is intentional to be able to update the material without retrieving it from the mesh every time.

Reply:
Thanks! :)